### PR TITLE
Make the columnsByKey variable reactive

### PR DESCRIPTION
**Pull request by dennisjlee**
Fixes dasDaniel/svelte-table#33

### CODE REVIEW

File: src/SvelteTable.svelte [hunk 1]
```diff
@@ -26,9 +26,13 @@
   let searchValues = {};
   let columnByKey = {};
 
-  columns.forEach(col => {
-    columnByKey[col.key] = col;
-  });
+  let columnByKey;
```
Comment (dasDaniel) [inline src/SvelteTable.svelte:29]: re-definition of `columnByKey` will cause errors

**Comment by dennisjlee** [comment]
Do you have any pointers on potential errors? I can do some further testing.

**Comment by dasDaniel** [comment]
The issue is only that `columnByKey` is defined twice; on line 27 and then on line 29

File: src/SvelteTable.svelte (hunk 1 above)
Comment (dennisjlee) [reply src/SvelteTable.svelte:27]: fixed

**Comment by dennisjlee** [comment]
@dasDaniel thanks!

**Pull request merged**